datafusion/core/src/config.rs

```diff
@@ -291,6 +291,10 @@ impl ConfigOptions {
     pub fn get_bool(&self, key: &str) -> bool {
         match self.get(key) {
             Some(ScalarValue::Boolean(Some(b))) => b,
+            Some(b) => b
```

Comment:
Thanks for looking at this @comphead. I think it might be better to do the conversion in the `set` methods rather than the `get` methods, and have the `set` methods return a `Result` but I haven't looked closely at this.

Reply:
Agreed with @andygrove. It would be better to make the configuration dynamically and strongly typed:
```rust
    pub fn get_bool(&self, key: &str) -> Result<bool> {
        match self.get(key) {
            Some(ScalarValue::Boolean(Some(b))) => b,
            Some(_) => Err(DatafusionError("The configuration {} is not a Boolean")),
            None => Err(DatafusionError("The configuration {} is not set")),
}}
```